proposed/simple-csharp-programs.md

```diff
@@ -0,0 +1,499 @@
+
+Console.WriteLine(person.Name);
+```
+
+And it'll run after downloading the package:
+
+```bash
+$ dotnet run
+dotnet bot
+```
+
+Or via a `#!` directive,
+
+```csharp
+#!/path/to/dotnet
+#r "nuget: Newtonsoft.Json"
+
+record Person(string Name);
+
+var json = "{\"name\": \"dotnet bot\"}";
+var person = JsonConvert.DeserializeObject<Person>(json);
+
+Console.WriteLine(person.Name);
+```
+
+and running:
+
+```bash
+$ ./hello.cs
+dotnet bot
+```
+
+The `#r "nuget:..."` syntax is already supported by C# today in .NET Interactive. It is a shared implementation between C# and F#. The syntax and mechanism could be re-used here.
+
+### Coding against a framework/sdk like ASP.NET Core
+
+You can also reference a framework/sdk (nomenclature up for debate) via a `#!` directive:
+
+```csharp
+#!/path/to/dotnet-aspnet
+
+var app = WebApplication.CreateBuilder(args);
+app.MapGet("/", () => "Hello World");
+await app.RunAsync();
+```
+
+The `dotnet-aspnet` gesture indicates two things to the build/run mechanism:
+
+* ASP.NET Core gets loaded and referenced
+* Applicable implicit ASP.NET Core `using`s are brought into scope
+
+And you can run it:
+
+```bash
+$ dotnet run
+Listening on localhost://12345
+```
+
+The syntax for coding against a framework/sdk could also be more similar to the `#r "nuget:"` syntax shown earlier instead of being a `#!` directive.
+
+### Arguments also work
+
+The following should be possible:
+
+```bash
+$ dotnet run arg1 arg2
+```
+
+or
+
+```bash
+$ ./hello.cs arg1 arg2`
+```
+
+### Running or building a specific file
+
+If you have multiple C# files in the same directory, `dotnet run` and `dotnet build` won't work by default. You'll need to specify the file you wish to run:
+
+```bash
+$ dotnet run hello.cs
+Hello World
+```
+
+or
+
+```bash
+$ dotnet build hello.cs
+```
+
+`dotnet build` could in theory work here, as it could just build an executable for each file. But symmetry with `dotnet run` is probably preferred.
+
+Multiple arguments should also work, but you'll still have to pass the file name:
+
+```bash
+$ dotnet run hello.cs arg1 arg2
+```
+
+### TFM chosen at build time
+
+For both `dotnet run` and `dotnet build`, a TFM to build against must be selected.
+
+It will be the latest TFM available in the installed SDK. So if you have the .NET 7 SDK, it will be built and run against `net7`.
+
+There is no ability to select a different TFM.
+
+## `dotnet` command support
+
+Only `dotnet run` and `dotnet build` are supported with Simple C# Programs. All other `dotnet` commands will error out, indicating that you need a project.
+
+## Synthesized project file
+
+Behind the scenes, a likely implementation will involved a synethesized project file. This would then have all of the information to make the `dotnet` commands work, and open us up to allowing more `dotnet` commands in the future.
+
+A synthezized project file could live either in memory or in the temp folder.
```

Comment:
I'm sure that'll come up in next round of discussion with SDK team: we'll want to specify what that project looks like.
I assume the `OutputType` would be `Exe`, for instance.